RecipeMatch/Views/GridRecipeCell.m

```diff
@@ -15,21 +15,30 @@ @implementation GridRecipeCell
 // clears image and title of cell
 - (void)prepareForReuse {
     [super prepareForReuse];
+    [self.imageView cancelImageDownloadTask];
+    self.imageView.image = NULL;
     [self.imageView sd_cancelCurrentImageLoad];
     self.recipeTitle.text = @"";
-    [self.searchImageView sd_cancelCurrentImageLoad];
-    self.searchRecipeTitle.text = @"";
 }
 
-- (void)setupWithRecipe:(SavedRecipe *)recipe {
+// sets up cell for profile page
+- (void)setupWithRecipeFromProfile:(SavedRecipe *)recipe {
+    // set outlets programmatically
+    self.imageView = (UIImageView *)[self viewWithTag:1];
+    self.recipeTitle = (UILabel *)[self viewWithTag:2];
+
     [self.imageView sd_setImageWithURL:[NSURL URLWithString:recipe.image] placeholderImage:nil];
     self.imageView.layer.cornerRadius = CORNER_RADIUS;
     self.recipeTitle.text = recipe.name;
 }
 
-- (void)searchSetupWithRecipe:(NSDictionary *)recipe {
-    [self.searchImageView sd_setImageWithURL:[NSURL URLWithString:recipe[@"image"]] placeholderImage:nil];
-    self.searchImageView.layer.cornerRadius = CORNER_RADIUS;
-    self.searchRecipeTitle.text = recipe[@"label"];
+// sets up cell for search page
+- (void)setupWithRecipeFromSearch:(NSDictionary *)recipe {
+    self.imageView = (UIImageView *)[self viewWithTag:3];
+    self.recipeTitle = (UILabel *)[self viewWithTag:4];
+
+    [self.imageView sd_setImageWithURL:[NSURL URLWithString:recipe[@"image"]] placeholderImage:nil];
+    self.imageView.layer.cornerRadius = CORNER_RADIUS;
+    self.recipeTitle.text = recipe[@"label"];
```

Comment:
not a fan of this approach....
it's better to a have custom reusable ui views (like this cell), as a separate class with a separate XIB file, and it's not a good practice to copy paste the same ui elements in different places in the storyboard and then find it in the code using `viewWithTag :`.
so for this PR, it's ok to leave this UI implementation as is. just maybe convert the tag numbers into constants.

Reply:
in addition, i'll ask to make some changes to how you setup this cell, as you can make the code more reusable and generic.
if you can't find a way to support one model for two tables with parse (it might be possible with inheritance?), then it would be better to setup the cell directly with a title, imageUrl and type.
something like:
``` 
// sets up cell for search page
- (void)setupWithRecipeTitle:(NSString *)recipeTitle recipeImageURl:(NSString *)recipeImageURl forProfileScreen:(BOOL)isForProfileScreen {
    self.imageView = (UIImageView *)[self viewWithTag:isForProfileScreen?1:3];
    self.recipeTitle = (UILabel *)[self viewWithTag:isForProfileScreen?2:4];

    [self.imageView sd_setImageWithURL:[NSURL URLWithString:recipeImageURl] placeholderImage:nil];
    self.imageView.layer.cornerRadius = CORNER_RADIUS;
    self.recipeTitle.text = recipeTitle;
}
 ``` 
(and even better have an enum instead of `forProfileScreen:(BOOL)forProfileScreen`)